source/MaterialXTest/XmlIo.cpp

```diff
@@ -40,6 +41,34 @@ TEST_CASE("Load content", "[xmlio]")
         libs.push_back(lib);
     }
 
+    // Check that there is one implementation per nodedef.
+    mx::DocumentPtr implCheckDocument = mx::createDocument();
+    for (mx::DocumentPtr lib : libs)
+    {
+        implCheckDocument->importLibrary(lib);
+    }
+    const std::string target;
+    const std::string language("osl");
+    std::vector<mx::NodeDefPtr> definitions = implCheckDocument->getNodeDefs();
```

Comment:
To reduce potential confusion between NodeDef and Implementation (either of which could lay claim to being the "definition" of a Node), I would recommend naming these variables "nodedefs" and "nodeDefsFound".